### PR TITLE
[ADD] Add module bank_statement_journal_items

### PR DESCRIPTION
**Pull request by ajaniszewska-dev**
Impacting module: Account

Add 'Journal items'  button to bank statement. 

cosanum1
cosanum2
cosanum3

### CODE REVIEW

**Review by vrenaville — APPROVED** [review]

File: bank_statement_journal_items/README.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Odoo bank statement journal items
+=================================
+
+This project allows to see Journal Items from bank statement.
```
Comment (Tonow-c2c) [inline bank_statement_journal_items/README.rst]: I thinks you should do a readme folder like you can see in https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

and after use the readme-generator https://github.com/OCA/maintainer-tools#readme-generator

**Review by Tonow-c2c — APPROVED** [review]
Code look good please change the readme

File: bank_statement_journal_items/models/bank_statement_items.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from odoo import models
+
+class BankStatementItems(models.Model):
```
Comment (grindtildeath) [inline bank_statement_journal_items/models/bank_statement_items.py]:
```suggestion
class AccountBankStatement(models.Model):
```
Please also rename the file to the model you're modifying (e.g. `account_bank_statement.py`)

File: bank_statement_journal_items/models/bank_statement_items.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from odoo import models
```
Comment (grindtildeath) [inline bank_statement_journal_items/models/bank_statement_items.py]: missing header

File: bank_statement_journal_items/models/bank_statement_items.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class BankStatementItems(models.Model):
+
+	_inherit="account.bank.statement"
+
+	def button_journal_items(self):
+		return {
+			'name': 'Journal Items',
```
Comment (grindtildeath) [inline bank_statement_journal_items/models/bank_statement_items.py]:
```suggestion
			'name': _('Journal Items'),
```

File: bank_statement_journal_items/models/bank_statement_items.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+			'view_mode': 'tree,form',
+			'res_model': 'account.move.line',
+			'views':[
+				(self.env.ref("account.view_move_line_tree").id, 'tree'),
+				(self.env.ref("account.view_move_line_form").id, 'form')
+			],
+			'type': 'ir.actions.act_window',
+			'domain': [('id', 'in', self.mapped('line_ids').ids)],
```
Comment (grindtildeath) [inline bank_statement_journal_items/models/bank_statement_items.py]:
```suggestion
			'domain': [('id', 'in', self.mapped('move_line_ids').ids)],
```
:warning: 
`line_ids` here will give you `account.bank.statement.line` records: https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_bank_statement.py#L196
Here we want to display `account.move.line`, so you must use `move_line_ids`: https://github.com/odoo/odoo/blob/13.0/addons/account/models/account_bank_statement.py#L197

File: bank_statement_journal_items/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Bank Statement Journal Entries",
```
Comment (grindtildeath) [inline bank_statement_journal_items/__manifest__.py]:
```suggestion
    "name": "Bank Statement Journal Items",
```

File: bank_statement_journal_items/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    "summary": "Add Journal Entries button in the bank statements",
```
Comment (grindtildeath) [inline bank_statement_journal_items/__manifest__.py]:
```suggestion
    "summary": "Add Journal Items button in the bank statements",
```

**Review by grindtildeath — CHANGES_REQUESTED** [review]

File: bank_statement_journal_items/README.rst (hunk 1 above)
Comment (ajaniszewska-dev) [reply bank_statement_journal_items/README.rst]: due to problems with dependencies i just created files under readme directory, without using readme-generator.

File: bank_statement_journal_items/readme/DESCRIPTION.rst [hunk 8]
```diff
@@ -0,0 +1 @@
+This module extends the functionality of bank statements and to allow you to see journal items list.
```
Comment (grindtildeath) [inline bank_statement_journal_items/readme/DESCRIPTION.rst]: Nitpicking but I would phrase it this way: "This module adds a smart button on bank statement form view, allowing you to display its journal items"

File: bank_statement_journal_items/readme/USAGE.rst [hunk 9]
```diff
@@ -0,0 +1,5 @@
+To use this module, you need to:
+
+1. Create Bank Statement.
+2. Click on Journal Items button in a bank statement.
+3. Click on Journal item from the list to see detailed information.
```
Comment (grindtildeath) [inline bank_statement_journal_items/readme/USAGE.rst]: I don't think usage makes sense if the description is spot on, but you could add yourself in `CONTRIBUTORS.rst` :wink:

File: bank_statement_journal_items/models/account_bank_statement.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.tools.translate import _
```
Comment (grindtildeath) [inline bank_statement_journal_items/models/account_bank_statement.py:5]:
```suggestion
from odoo import _, models
```
Nitpicking, but you can import `_` directly from `odoo`

**Review by grindtildeath — APPROVED** [review]

**Review by guewen — APPROVED** [review]

**Pull request merged**